File: go/src/day1/puzzle.go
package day1

func SolvePart1(input []int) int {
	sum := 0

	for i := 0; i < len(input); i++ {
		current := input[i]
		next := input[(i + 1) % len(input)]

		if current == next {
			sum += current
		}
	}

	return sum
}

func SolvePart2(input []int) int {
	sum := 0

	for i := 0; i < len(input); i++ {
		current := input[i]
		halfwayAround := input[(i + len(input) / 2) % len(input)];

		if current == halfwayAround {
			sum += current
		}
	}

	return sum
}

File: 2017/dotnet/src/Day1/Puzzle.cs
namespace AdventOfCode.Day1
{
    public static class Puzzle
    {
        public static int SolvePart1(int[] input)
        {
            var sum = 0;

            for (var i = 0; i < input.Length; i++)
            {
                var current = input[i];
                var next = input[(i + 1) % input.Length];

                if (current == next)
                {
                    sum += current;
                }
            }

            return sum;
        }

        public static int SolvePart2(int[] input)
        {
            var sum = 0;

            for (var i = 0; i < input.Length; i++)
            {
                var current = input[i];
                var halfwayAround = input[(i + input.Length / 2) % input.Length];

                if (current == halfwayAround)
                {
                    sum += current;
                }
            }

            return sum;
        }
    }
}
